app/client/components/payment/update/confirmPaymentUpdate.tsx

```diff
@@ -0,0 +1,155 @@
+        loadingMessage={`Updating ${
+          this.props.newPaymentMethodDetail.friendlyName
+        } details...`}
+        spinnerScale={0.7}
+        inline
+      />
+    ) : (
+      <Button
+        text={CONFIRM_BUTTON_TEXT}
+        onClick={() => this.setState({ hasHitComplete: true })}
+        primary
+        right
+      />
+    );
+  }
+
+  private executePaymentUpdate: () => Promise<Response> = async () =>
+    await fetch(
+      `/api/payment/${this.props.productType.urlPart}/${
+        this.props.newPaymentMethodDetail.apiUrlPart
+      }`,
+      {
+        credentials: "include",
+        method: "POST",
+        body: JSON.stringify(
+          this.props.newPaymentMethodDetail.detailToPayloadObject()
+        ),
+        headers: { "Content-Type": "application/json" }
+      }
+    );
+
+  private renderUpdateResponse = (response: object) => {
+    if (
+      this.props.navigate &&
+      this.props.newPaymentMethodDetail.matchesResponse(response)
+    ) {
+      trackEvent({
+        eventCategory: "payment",
+        eventAction: this.props.newPaymentMethodDetail.name + "_update_success",
+        eventLabel: this.props.productType.urlPart
+      });
+      this.props.navigate("updated", { replace: true });
+      return null;
+    }
+
+    return this.PaymentUpdateFailed();
+  };
+
+  private PaymentUpdateFailed = () => {
+    trackEvent({
+      eventCategory: "payment",
+      eventAction: this.props.newPaymentMethodDetail.name + "_update_failed",
+      eventLabel: this.props.productType.urlPart
+    });
+
+    Raven.captureException(
+      this.props.newPaymentMethodDetail.friendlyName + "payment update failed"
+    );
+
+    return (
+      <div css={{ textAlign: "left", marginTop: "10px" }}>
+        <h2>
+          Sorry, the {this.props.newPaymentMethodDetail.friendlyName} update
+          failed.
+        </h2>
+        <p>
+          To try again please go back and re-enter your new{" "}
+          {this.props.newPaymentMethodDetail.friendlyName} details.
+          Alternatively, please call to speak to one of our customer service
+          specialists.
+        </p>
+        <CallCentreNumbers prefixText="To contact us" />
+      </div>
+    );
+  };
+}
+
+export const ConfirmPaymentUpdate = (props: RouteableStepProps) => (
+  <NewPaymentMethodContext.Consumer>
+    {newPaymentMethodDetail => (
+      <MembersDataApiResponseContext.Consumer>
+        {mdaResponse =>
+          props.navigate &&
+          isNewPaymentMethodDetail(newPaymentMethodDetail) &&
+          hasProduct(mdaResponse) ? (
+            <WizardStep
+              routeableStepProps={labelPaymentStepProps(props)}
+              extraFooterComponents={<QuestionsFooter />}
+            >
+              <h3>Please confirm your change from...</h3>
+              <CurrentPaymentDetails {...mdaResponse.subscription} />
+              <h3>...to...</h3>
+              {newPaymentMethodDetail.render()}
```

Comment:
Out of interest only - and maybe I need that tutorial in React - why is this an imported module with newPaymentMethodDetail.render() and not just a nested custom JSX tag? e.g. <NewPaymentMethodDetail /> ?

Reply:
So the imported module `NewPaymentMethodDetail` just defines the type/interface. I happen to have named the variable (from the consumer) the same thing, but note the  lower case starting letter (as per convention), it just so happens to a have method on that type called `render` but that could be changed... It isn't actually a 'Component' and hence couldn't be displayed by using custom JSX.

Reply:
`NewPaymentMethodDetail` could be rewritten as an extension of Component and hence could be instantiated/called by adding a JSX tag, but that could be misleading I think.